### PR TITLE
executor: fix wrong row count in fast analyze

### PR DESCRIPTION
**Pull request by lamxTyler**
### What problem does this PR solve? 

```
FAIL: analyze_test.go:400: testFastAnalyze.TestFastAnalyzeRetryRowCount
analyze_test.go:421:
    c.Assert(row[5], Equals, "30")
... obtained string = "0"
... expected string = "30"
```

### What is changed and how it works?

When stats lease is not 0, the row count of fast analyze result is `mathutil.MinInt64(stats.GetTableStats(e.tblInfo).Count, rowCount)`

- If the test runs fast, the stats cache has not been updated, so the `Count` will be pseudo row count 10000, so we can get the expected result.
- If the test runs slow, newly added rows are not dumped to stats table yet, so the `Count` in stats cache will be `0`, and the test will fail.

This PR fixes it by also consider the sampled count when calculating the result count.

### Check List 

Tests 

- Unit test

Code changes

 - Has exported function/method change

Side effects

 - None

Related changes

 - None

### CODE REVIEW

**Comment by zz-jason** [comment]
I think if `stats.GetTableStats(e.tblInfo).Pseudo` is true, we should not execute:
```go
rowCount = mathutil.MinInt64(stats.GetTableStats(e.tblInfo).Count, rowCount)
```

File: executor/analyze.go [hunk 1]
```diff
@@ -990,11 +990,13 @@ func (e *AnalyzeFastExec) buildHist(ID int64, collector *statistics.SampleCollec
...
-	stats := domain.GetDomain(e.ctx).StatsHandle()
+	handle := domain.GetDomain(e.ctx).StatsHandle()
+	tblStats := handle.GetTableStats(e.tblInfo)
 	rowCount := int64(e.rowCount)
-	if stats.Lease() > 0 {
-		rowCount = mathutil.MinInt64(stats.GetTableStats(e.tblInfo).Count, rowCount)
+	if handle.Lease() > 0 && !tblStats.Pseudo {
+		rowCount = mathutil.MinInt64(tblStats.Count, rowCount)
 	}
+	rowCount = mathutil.MaxInt64(rowCount, int64(len(collector.Samples)))
```
Comment (XuHuaiyu) [inline executor/analyze.go]: Should we add some comments for why we choose this rowCount?

**Pull request merged**